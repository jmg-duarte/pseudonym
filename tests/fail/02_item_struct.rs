use pseudonym::alias;

#[alias(
    S1,
    deprecated(S2),
    deprecated(S3, since = "0.1.0"),
    deprecated(S4, note = "deprecation note"),
    deprecated(S5, since = "0.1.0", note = "deprecation note")
)]
struct S(i32);

fn main() {
    assert_eq!(S(5).0, 5);
    assert_eq!(S1(5).0, 5);
    assert_eq!(S2(5).0, 5);
    assert_eq!(S3(5).0, 5);
    assert_eq!(S4(5).0, 5);
    assert_eq!(S5(5).0, 5);
}
