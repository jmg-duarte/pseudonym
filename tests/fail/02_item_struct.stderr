error: unexpected end of input, expected identifier
 --> tests/fail/02_item_struct.rs:3:1
  |
3 | #[alias(S1, S2,)]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `alias` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find function, tuple struct or tuple variant `S` in this scope
 --> tests/fail/02_item_struct.rs:7:16
  |
7 |     assert_eq!(S(5).0, 5);
  |                ^ not found in this scope

error[E0425]: cannot find function, tuple struct or tuple variant `S1` in this scope
 --> tests/fail/02_item_struct.rs:8:16
  |
8 |     assert_eq!(S1(5).0, 5);
  |                ^^ not found in this scope

error[E0425]: cannot find function, tuple struct or tuple variant `S2` in this scope
 --> tests/fail/02_item_struct.rs:9:16
  |
9 |     assert_eq!(S2(5).0, 5);
  |                ^^ not found in this scope
